### PR TITLE
Check and warn for falsy class name (fixes #4050)

### LINKED ISSUE #4050: Transition class bind null values has DOMException in addClass function.
### Vue.js version

2.0.1

```
<transition :enter-active-class="animatedIn" >
```

When 'animateIn' bind value is empty string or null,addClass function has exception:

```
vue.js:4540 Uncaught (in promise) DOMException: Failed to execute 'add' on 'DOMTokenList': The token provided must not be empty
```

Semantically,Binding a null value is intended to remove the transition effect.
Should a null check be made in addClass function or the transition should be ignored in the case of null values?

### PR DESCRIPTION
**Pull request by phanan**
This commit adds a check for falsy names (null or empty string) before attempting to add or remove them, to prevent a DOM exception. A warning will also be triggered if in development env.

### CODE REVIEW

File: src/platforms/web/runtime/class-util.js [hunk 1]
```diff
@@ -25,6 +32,11 @@ export function addClass (el: Element, cls: string) {
  * SVG elements in IE
  */
 export function removeClass (el: Element, cls: string) {
```
Comment (phanan) [inline src/platforms/web/runtime/class-util.js]: You're right, I totally missed this. Adding it in, though maybe we don't need to cater for `null` or `undefined` at all (a "`.trim()` is not a function" error will be thrown).

**Comment by posva** [comment]
@phanan Just to make it clear: this won't fix what the feature he was asking for. He was asking the class to be removed when the value is null. Or did I misunderstood something?

**Comment by phanan** [comment]
No, it won't really "fix" what he asked for, but the DOM exception. The semantic idea (`null` removes the effect) is left undecided, though personally I'm not a fan of it.

**Comment by sean220** [comment]
So,Could you provide a better way to cancel the transition effect dynamically? @phanan

**Pull request merged**